### PR TITLE
write proguard uuid to manifest

### PR DESCRIPTION
**Pull request by lbloder**
## :scroll: Description
Write proguard uuid to AndroidManifest

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryGenerateProguardUuidTask.kt [hunk 1]
```diff
@@ -2,35 +2,50 @@ package io.sentry.android.gradle.tasks
...
     @get:Internal
     val outputFile: Provider<RegularFile> get() = output.file(SENTRY_UUID_OUTPUT)
 
     @TaskAction
     fun generateProperties() {
-        val outputDir = output.get().asFile
-        outputDir.mkdirs()
+        val outputDir = output.orNull?.asFile
+
+        if (outputDir != null) {
```
Comment (adinauer) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryGenerateProguardUuidTask.kt:36]: Does this cause problems where the file is no longer available for other tasks that rely on it, like `SentryUploadProguardMappingsTask`?

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/AndroidComponentsConfig.kt [hunk 2]
```diff
@@ -164,21 +165,34 @@ private fun Variant.configureProguardMappingsTasks(
...
         val guardsquareEnabled = extension.experimentalGuardsquareSupport.get()
         val isMinifyEnabled = isMinificationEnabled(project, variant, guardsquareEnabled)
 
         if (isMinifyEnabled && extension.includeProguardMapping.get()) {
-            val generateUuidTask =
-                SentryGenerateProguardUuidTask.register(
-                    project = project,
-                    taskSuffix = name.capitalized
-                )
             configureGeneratedSourcesFor74(
```
Comment (adinauer) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/AndroidComponentsConfig.kt:193]: Should this be skipped, when manifest is used?

**Comment by romtsn** [comment]
@lbloder sorry for holding this off again. We've been discussing internally if we would like to drop support for AndroidManifest, because we're not sure if it's used a lot given lack of documentation for options, but also because parsing the manifest takes almost half-the-time of Sentry.init as we recently figured out. I'd keep this PR open until we come up to a conclusion and then take care of it accordingly.

Sorry for back-and-forths again!

**Comment by adinauer** [comment]
@romtsn can we close this? We can come back and reopen if we still need it.

**Comment by adinauer** [comment]
Closing, if we need this we can reopen.

**Pull request closed without merging**